packages/server/graphql/public/mutations/createStripeSubscription.ts

```diff
@@ -38,14 +38,18 @@ const createStripeSubscription: MutationResolvers['createStripeSubscription'] =
     return standardError(new Error('Organization already has a subscription'), {userId: viewerId})
   }
   const {email} = viewer
-  const customer = stripeId
-    ? await manager.retrieveCustomer(stripeId)
-    : await manager.createCustomer(orgId, email)
-  const {id: customerId} = customer
-  const res = await manager.attachPaymentToCustomer(customerId, paymentMethodId)
-  if (res instanceof Error) return standardError(res, {userId: viewerId})
-  // wait until the payment is attached to the customer before updating the default payment method
-  await manager.updateDefaultPaymentMethod(customerId, paymentMethodId)
+  let customer: Stripe.Response<Stripe.Customer | Stripe.DeletedCustomer>
+  if (stripeId) {
+    customer = await manager.retrieveCustomer(stripeId)
+    const {id: customerId} = customer
+    const res = await manager.attachPaymentToCustomer(customerId, paymentMethodId)
+    if (res instanceof Error) return standardError(res, {userId: viewerId})
+    // wait until the payment is attached to the customer before updating the default payment method
+    await manager.updateDefaultPaymentMethod(customerId, paymentMethodId)
+  } else {
+    customer = await manager.createCustomer(orgId, email, paymentMethodId)
+  }
+
```

Comment:
And we could remove the `await` on line 65 when updating rethinkdb?

Reply:
i was thinking about that! but since our DB is colocated it's usually only a couple ms so I figured probably better to be safe than sorry in case the write failed for some reason.

Reply:
On second look, we should completely get rid of the write to the DB here! 
We should only update our DB when the payment succeeds, otherwise if we update the stripeSubscriptionId & then the 3D payment fails, the org gets stuck in a weird state where it can't upgrade because it has a subscription, but it can't downgrade because it's still on the starter tier 😬